src/Raven.Client/Documents/Commands/Batches/BatchCommand.cs

```diff
@@ -15,37 +15,46 @@ namespace Raven.Client.Documents.Commands.Batches
 {
     internal class ClusterWideBatchCommand : SingleNodeBatchCommand, IRaftCommand
     {
+        public bool? DisableAtomicDocumentWrites { get; }
         public string RaftUniqueRequestId { get; } = RaftIdGenerator.NewId();
 
-        public ClusterWideBatchCommand(DocumentConventions conventions, JsonOperationContext context, IList<ICommandData> commands, BatchOptions options = null) : base(conventions, context, commands, options, TransactionMode.ClusterWide)
+        public ClusterWideBatchCommand(DocumentConventions conventions, IList<ICommandData> commands, BatchOptions options = null, bool? disableAtomicDocumentsWrites = null) : base(conventions, commands, options, TransactionMode.ClusterWide)
         {
+            DisableAtomicDocumentWrites = disableAtomicDocumentsWrites;
+        }
+
+        protected override void AppendOptions(StringBuilder sb)
+        {
+            base.AppendOptions(sb);
+            if (DisableAtomicDocumentWrites == null)
+                return;
+            sb.Append("&disableAtomicDocumentWrites=")
+                .Append(DisableAtomicDocumentWrites.Value ? "true" : "false");
         }
     }
 
     public class SingleNodeBatchCommand : RavenCommand<BatchCommandResult>, IDisposable
     {
-        private readonly BlittableJsonReaderObject[] _commands;
+        private BlittableJsonReaderObject[] _commandsAsJson;
+        private bool? _supportsAtomicWrites;
         private readonly List<Stream> _attachmentStreams;
         private readonly HashSet<Stream> _uniqueAttachmentStreams;
+        private readonly JsonOperationContext _context;
+        private readonly DocumentConventions _conventions;
+        private readonly IList<ICommandData> _commands;
         private readonly BatchOptions _options;
         private readonly TransactionMode _mode;
 
-        public SingleNodeBatchCommand(DocumentConventions conventions, JsonOperationContext context, IList<ICommandData> commands, BatchOptions options = null, TransactionMode mode = TransactionMode.SingleNode)
+        public SingleNodeBatchCommand(DocumentConventions conventions,  IList<ICommandData> commands, BatchOptions options = null, TransactionMode mode = TransactionMode.SingleNode)
```

Comment:
breaking change - we need to revert this

Reply:
done